chia-protocol/src/bytes.rs

```diff
@@ -159,6 +169,14 @@ impl<const N: usize> fmt::Display for BytesImpl<N> {
     }
 }
 
+#[cfg(feature = "serde")]
+impl<const N: usize> Serialize for BytesImpl<N> {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        let s = format!("0x{}", hex::encode(&self.0));
+        serializer.serialize_str(&s)
```

Comment:
this is a bit of a hack. it will make the JSON serializer do the right thing, but any other serde serializer will do the wrong thing. which may not be the end of the world, since we don't need any other one, but it might also be surprising.

Reply:
Yes, this is a bit of a hack. I'll have a look if I can limit the specialisation to just serde_json's Serializer and let every other serde serializer just serialize as bytes.

Reply:
Nope, still no way, afaict.